cypher/cypher-docs/src/docs/dev/deprecations-additions-and-compatibility.asciidoc

```diff
@@ -103,29 +103,148 @@ a|
 Token lookup indexes now allow an `OPTIONS` map to specify the index provider.
 
 
+a|
+label:functionality[]
+label:added[]
+[source, cypher, role="noheader"]
+----
+CREATE TEXT INDEX ...
+----
+a|
+Allows creating text indexes on nodes or relationships with a particular label or relationship type and property combination.
+They can be dropped by using their name.
+
+
+a|
+label:functionality[]
+label:added[]
+[source, cypher, role="noheader"]
+----
+CREATE RANGE INDEX ...
+----
+a|
+Allows creating range indexes on nodes or relationships with a particular label or relationship type and properties combination.
+They can be dropped by using their name.
+
+
+a|
+label:functionality[]
+label:added[]
+[source, cypher, role="noheader"]
+----
+CREATE CONSTRAINT
+...
+OPTIONS "{" indexProvider: 'range-1.0' "}"
+----
+a|
+Allows creating node key and uniqueness constraints backed by range indexes by providing the range index provider in the `OPTIONS` map.
+
+
+a|
+label:functionality[]
+label:added[]
+[source, cypher, role="noheader"]
+----
+CREATE POINT INDEX ...
+----
+a|
+Allows creating point indexes on nodes or relationships with a particular label or relationship type and property combination.
+They can be dropped by using their name.
+
 
 a|
 label:functionality[]
 label:deprecated[]
 [source, cypher, role="noheader"]
 ----
-CREATE BTREE INDEX ...
+CREATE [BTREE] INDEX
+...
+OPTIONS "{" indexProvider: 'lucene+native-3.0' "}"
 ----
 a|
 B-tree indexes are deprecated and need to be migrated before upgrading to 5.0.
 
+B-tree indexes with the `lucene+native-3.0` index provider and no index configuration settings are replaced by:
+[source, cypher, role="noheader"]
+----
+CREATE TEXT INDEX ...
+----
+
 a|
 label:functionality[]
 label:deprecated[]
 [source, cypher, role="noheader"]
 ----
-CREATE INDEX
+CREATE [BTREE] INDEX
 ...
-OPTIONS "{" btree-option: btree-value[, ...] "}"
+OPTIONS "{" indexProvider: 'lucene+native-3.0', indexConfig: btree-spatial-setting[, ...] "}"
+----
+a|
+B-tree indexes are deprecated and need to be migrated before upgrading to 5.0.
```

Comment:
There have been more discussions about migration strategy but this is still likely. Can always be changed later if needs too.
But if planner is not done these should maybe not go in docs at all yet and we add them when closing in on 5.0 or something

Reply:
None of the documentation prs will be merged unless there is planner support